### PR TITLE
[HTTP/2.0][WIP] H2 server documentation

### PR DESCRIPTION
**Pull request by dheiberg**
I've written some documentation for the majority of the H2ResponseWriter API aswell as a general overview of how the new frame handlers work and how to use them. Still need to cover a few more functions in the API. Otherwise I'm not quite sure what is missing, thoughts?

### CODE REVIEW

File: docs/_writing-tests/server-features.md [hunk 1]
```diff
@@ -94,4 +94,19 @@ of the response. For details see the
 [wptserve documentation](https://wptserve.readthedocs.org).
 
 
+### Writing tests for HTTP/2.0
+
+The server now has a prototype HTTP/2.0 server which gives you access to 
+some of the HTTP/2.0 specific functionality. Currently, the server is off
+by default and needs to be run using `./wpt serve --h2` in order to enable it.
+To find out more about the new Python handlers introduced head over to
```
Comment (jgraham) [inline docs/_writing-tests/server-features.md]: "head over to" seems a little colloquial. I would say something like "The HTTP 2 server supports handlers that work per-frame; these are documented in […]".

File: docs/_writing-tests/h2responsewriter.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
+## Response Writer API ##
+
+##### `write_headers(self, headers, status_code, status_message=None, stream_id=None, last=False):`
+Send a HEADER frame that is tracked by the local state machine.
```
Comment (jgraham) [inline docs/_writing-tests/h2responsewriter.md]: I'm not sure these one sentence summaries of the two sentence descriptions are adding much.

File: docs/_writing-tests/h2responsewriter.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
+## Response Writer API ##
+
+##### `write_headers(self, headers, status_code, status_message=None, stream_id=None, last=False):`
```
Comment (jgraham) [inline docs/_writing-tests/h2responsewriter.md]: This header level seems very low; is there prededent somewhere else in the docs?

**Review by jgraham — COMMENTED** [review]
Generally this seems good, just some minor comments.

File: docs/_writing-tests/h2responsewriter.md (hunk 3 above)
Comment (dhdavvie) [reply docs/_writing-tests/h2responsewriter.md]: As far as I can see, there is no precedent for this kind of API documentation. There are some js API docs but those are mostly showing a block of code and explaining what they do by the looks of it. I can bump it up a level or two?

Comment (jgraham) [reply docs/_writing-tests/h2responsewriter.md]: Yeah, see if that looks better.

**Comment by annevk** [comment]
Looks great, thanks!

docs/_writing-tests/h2responsewriter.md should probably have a pointer back to the main API discussion, or maybe be inlined there? An example or two for that section would also be nice.

**Comment by dhdavvie** [comment]
@annevk do you mean the API discussion that is open on GitHub?

**Comment by annevk** [comment]
No, I meant to docs/_writing-tests/h2tests.md.

**Comment by dhdavvie** [comment]
Ahh thought so, just checking :) I think inlining the responsewriter API into h2tests.md would make a lot of sense, @jgraham thoughts?

**Comment by jgraham** [comment]
Yeah, that could work.

**Review by jgraham — APPROVED** [review]

**Pull request merged**